cmd/ci-operator-prowgen/main.go

```diff
@@ -263,7 +264,13 @@ func extractRepoElementsFromPath(configFilePath string) (*configFilePathElements
 	}
 
 	fileName := filepath.Base(configFilePath)
-	branch := strings.TrimSuffix(fileName, filepath.Ext(configFilePath))
+	s := strings.Split(strings.TrimSuffix(fileName, filepath.Ext(configFilePath)), "-")
+
+	branch := s[len(s)-1]
+
+	if _, err := strconv.ParseFloat(branch, 64); err == nil {
+		branch = s[len(s)-2] + "-" + s[len(s)-1]
+	}
```

Comment:
This will still break on branch names using `-` sign where the last part is not numeric. Like `openshift-master`, `whatever-testing` etc.